cantools/typechecking.py

```diff
@@ -1,27 +1,60 @@
-import typing
+from typing import (
+    TYPE_CHECKING,
+    NamedTuple,
+    Union,
+    Dict,
+    Optional,
+    List,
+    Any,
+    Sequence,
+    Tuple,
+)
 
-import typing_extensions
+from typing_extensions import TypedDict
 from bitstruct import CompiledFormatDict
 
-if typing.TYPE_CHECKING:
+
+if TYPE_CHECKING:
     import os
     import sys
-    from cantools.database import Signal
+    from cantools.database import Signal, Message
+    from cantools.database.can.signal import NamedSignalValue
 
 
-class Formats(typing.NamedTuple):
+class Formats(NamedTuple):
     big_endian: CompiledFormatDict
     little_endian: CompiledFormatDict
     padding_mask: int
 
 
-StringPathLike = typing.Union[str, "os.PathLike[str]"]
-Comments = typing.Dict[typing.Optional[str], str]
-Codec = typing_extensions.TypedDict(
+StringPathLike = Union[str, "os.PathLike[str]"]
+Comments = Dict[Optional[str], str]
+Codec = TypedDict(
     "Codec",
     {
-        "signals": typing.List["Signal"],
+        "signals": List["Signal"],
         "formats": Formats,
-        "multiplexers": typing.Dict[str, typing.Dict[int, typing.Any]],
+        "multiplexers": Dict[str, Dict[int, Any]],  # "Any" should be "Codec" (cyclic definition is not possible though)
     },
 )
+
+
+# Type aliases. Introduced to reduce type annotation complexity while
```

Comment:
nice. I missed that file (`typechecking.py`) so  I put the type aliases into `message.py`. Thanks for cleaning up after me :)